design/common/multi-net-multus.md

```diff
@@ -0,0 +1,110 @@
+`cluster` are specified for Ceph.
+
+### Network Interface Selector
+
+Network interface selector is fed to network provider to connect pod to cluster
+network. This selector may vary from one network provider to another. For
+example, host network provider only needs to know the interface name.
+
+On the other hand, multi-plugin CNI plugin needs to know the network attachment
+definition's name and vice versa. Multi-plugin such as multus may seem to follow
+_Network Attachment Selection Annotation_ documented by Kubernetes Network Plumbing Working Group's
+[Multi-Net Spec].
+
+Any future multi-homed network provider that implements the Network Plumbing WG's Multi-net spec may
+use the `multus` network. Previously-identified providers [CNI-Genie] and [Knitter] have since gone
+dormant (upd. 2023), and Multus continues to be the most prominent known provider.
+
+## Multi-Net validation tester
+
+Rook will implement a user-runnable routine that validates a multi-net configuration based on the
+[Multi-Net Spec]. Users can input one or both of the Network Attachment Definitions (NADs) for
+Ceph's `multus` network, and the routine will perform cursory validation of the ability of the
+specified networks to support Ceph.
+
+The routine will start up **a single** web server with the specified networks attached (via NADs).
+The routine will also start up a number of clients on each node that will test the network(s)'s
+connections by HTTP(S) requests to the web server.
+
+A client represents a simplified view of a Ceph daemon from the networking perspective. The number
+of clients is intended to represent the number of Ceph daemons that could run on a node in the worst
+possible failure case where all daemons are rescheduled to a single node. This helps verify that the
+network provider is able to successfully assign IP addresses to Ceph daemon pods in the worst case.
+Some network interface software/hardware may limit the number of addresses that can be assigned to
+an interface on a given node, and this helps verify that issue is not present.
+
+It is important that the web server be only **a single** instance. If clients collocated on the same
+node with the web server can make successful HTTP(S) requests but clients on other nodes are not,
+that is an important indicator of inter-node traffic being blocked.
+
+The validation test routine will run in the Rook operator container. Rook's kubectl [Krew plugin]
+may facilitate running the test, but it is useful to have the option of executing a long-running
+validation routine running in the Kubernetes cluster instead of on an administrator's log-in
+session. Additionally, the results of the validation tester may become important for users wanting
+to get help with Multus configuration, and having a tool that is present in the Rook container image
+will allow users to run the routine for bug reports more readily than if they needed to install the
+Krew plugin.
+
```

Comment:
Some design questions:
- The operator already has all the needed RBAC, right, So no additional privileges are needed just for this tool?
- We don't see a need to support running the tool outside the operator pod, right? That's what I'm understanding from this paragraph, but I thought from earlier discussions perhaps there was a need to run it independently. But for simplicity I certainly vote for just running it in the operator at least until a specific requirement comes up.

Reply:
1. Correct, although the privileges needed for the tool are pretty minimal already. It already meets the security requirements of most reasonably strict pod security standards. Any user or serviceaccount that can create/get/list/delete configmaps, pods, and daemonsets can run the tool.
2. I suppose that's a fair point. I have been running it independently, and I have been sending beta builds to a couple of users, but we can certainly limit our support to running it in the operator pod for upstream. IMO, it probably won't be a big risk either way, but it is usually easier to have just one way of accomplishing things. 

I will plan to update the doc to only mention running within the operator pod. I will also update multus docs to mention the tool's existence.